### ISSUE TITLE
ResponseBuilder::streamed_body not streaming properly

### ISSUE THREAD

**Issue by Wazzaps** [SergioBenitez/Rocket#1389]
I'm trying to stream a list of strings which I get from a `mpsc`, but all of the results appear at once.
Illustration (from Chrome devtools):
![image](https://user-images.githubusercontent.com/6624767/88452724-bd3cfc80-ce69-11ea-809a-08458faf68ce.png)
I expect the results to stream, line by line.
I found no info about this in the [ResponseBuilder::streamed_body](https://api.rocket.rs/v0.4/rocket/response/struct.ResponseBuilder.html#method.streamed_body) or [Response::set_streamed_body](https://api.rocket.rs/v0.4/rocket/struct.Response.html#method.set_streamed_body).

I also tried to use `chunked_body` with size 1, that didn't help.

## Info
1. **Rocket Version:** 0.4.5
2. **OS:** Pop_OS! 20.04 (Based on Ubuntu 20.04)
3. **Kernel:** `Linux version 5.4.0-7634-generic (buildd@lgw01-amd64-017) (gcc version 9.3.0 (Ubuntu 9.3.0-10ubuntu2)) #38~1592497129~20.04~9a1ea2e-Ubuntu SMP Fri Jun 19 22:43:37 UTC`

## Code: main.rs
```rust
#[get("/")]
fn example() -> Option<StreamedStringListResponse> {
    let (send, recv) = std::sync::mpsc::channel();
    std::thread::spawn(move || {
        for i in 0..4 {
            send.send(format!("{}\n", i)).unwrap();
            sleep(Duration::from_millis(500));
        }
    });
    Some(StreamedStringListResponse::new(recv))
}

fn main() {
    autocomplete::init();
    rocket::ignite()
        .mount("/", routes![example])
        .launch();
}
```

## Code: streamed_string_list_response.rs
```rust
pub struct StreamedStringListResponse {
    source: Receiver<String>,
    cursor: Option<Cursor<String>>,
}

// Source: https://github.com/SergioBenitez/Rocket/issues/1298#issuecomment-629516104
impl Read for StreamedStringListResponse {
    fn read(&mut self, buf: &mut [u8]) -> Result<usize, std::io::Error> {
        if self.cursor.is_none() {
            // Get the next chunk
            let chunk = match self.source.iter().next() {
                Some(f) => f,
                None => {
                    println!("Done");
                    return Ok(0);
                },
            };

            self.cursor = Some(Cursor::new(chunk));
        }

        // Read the current chunk into the buffer
        let cursor = self.cursor.as_mut().unwrap();
        let n = cursor.read(buf);
        // If we completed the chunk, reset `cursor` to `None` so that
        // the next chunk will be taken the next time `read` is called
        if cursor.position() == cursor.get_ref().len() as u64 {
            self.cursor = None;
        }
        // <TEMP>
        if n.is_err() {
            println!("err");
        }
        let n = n.unwrap();
        println!("returning {} bytes", n);
        // </TEMP>
        Ok(n)
    }
}

impl StreamedStringListResponse {
    pub fn new(source: Receiver<String>) -> Self {
        StreamedStringListResponse {
            source,
            cursor: None,
        }
    }
}

impl<'r> Responder<'r> for StreamedStringListResponse {
    fn respond_to(self, _: &Request) -> response::Result<'r> {
        Response::build()
            .header(ContentType::Plain)
            .streamed_body(self)
            .ok()
    }
}
```

## Output
```
(...typical rocket output...)
read
returning 3 bytes
read
returning 3 bytes
read
returning 3 bytes
read
returning 3 bytes
read
returning 3 bytes
read
Done
read
Done
read
```

The "read" lines _are_ streamed as expected, which makes me think this is a Rocket problem.
Also, the fact we read twice after it's done is curious.

**Comment by jebrosen** [comment]
I think this is the same issue addressed by #1365 - i.e. that there isn't currently an API to force an early flush of the response data.

**Comment by SergioBenitez** [comment]
Indeed, this is a known, by-design issue, which may be resolved by #1365. Let's track this there.
